numba/tests/test_remove_dead.py

```diff
@@ -78,8 +91,97 @@ def seed_call_exists(func_ir):
             return False
 
         test_ir = compiler.run_frontend(call_np_random_seed)
-        remove_dead(test_ir.blocks, test_ir.arg_names)
+        remove_dead(test_ir.blocks, test_ir.arg_names, test_ir)
         self.assertTrue(seed_call_exists(test_ir))
 
+    def run_array_index_test(self, func):
+        A1 = np.arange(6).reshape(2,3)
+        A2 = A1.copy()
+        i = 0
+        pfunc = self.compile_parallel(func, (numba.typeof(A1), numba.typeof(i)))
+
+        func(A1, i)
+        pfunc(A2, i)
+        np.testing.assert_array_equal(A1, A2)
+
+    def test_alias_ravel(self):
+        def func(A, i):
+            B = A.ravel()
+            B[i] = 3
+
+        self.run_array_index_test(func)
+
+    def test_alias_flat(self):
+        def func(A, i):
+            B = A.flat
+            B[i] = 3
+
+        self.run_array_index_test(func)
+
+    def test_alias_transpose1(self):
+        def func(A, i):
+            B = A.T
+            B[i,0] = 3
+
+        self.run_array_index_test(func)
+
+    def test_alias_transpose2(self):
+        def func(A, i):
+            B = A.transpose()
+            B[i,0] = 3
+
+        self.run_array_index_test(func)
+
+    def test_alias_transpose3(self):
+        def func(A, i):
+            B = np.transpose(A)
+            B[i,0] = 3
+
+        self.run_array_index_test(func)
+
+    @needs_blas
+    def test_alias_ctypes(self):
+        # use xxnrm2 to test call a C function with ctypes
+        from numba.targets.linalg import _BLAS
+        xxnrm2 = _BLAS().numba_xxnrm2(types.float64)
+
+        def remove_dead_xxnrm2(rhs, lives, call_list):
+            if call_list == [xxnrm2]:
+                return rhs.args[4].name not in lives
+            return False
+
+        # adding this handler has no-op effect since this function won't match
+        # anything else but it's a bit cleaner to save the state and recover
```

Comment:
Should this be in a `try...finally`, if the body of the test fails, the state recovery still needs to run?

Reply:
done.